pepper-apis/src/main/java/org/broadinstitute/ddp/model/event/activityinstancecreation/ActivityInstanceCreatorFromAnswers.java

```diff
@@ -0,0 +1,165 @@
+
+            // detect number of new activities to be created
+            int newInstancesCount = sourceAnswers.size();
+            if (newInstancesCount < numberOfActivitiesLeft) {
+                newInstancesCount = numberOfActivitiesLeft;
+                LOG.warn("Only {} activity instances can be created (out of {}) because limit of instances is reached",
+                        newInstancesCount, sourceAnswers.size());
+            }
+
+            creationService.hideExistingInstancesIfRequired(studyActivityId, handle, activityDto);
+
+            // create child instances and copy source answers to target question in the created activities
+            for (int i = 0; i < newInstancesCount; i++) {
+
+                long newActivityInstanceId =
+                        createActivityInstance(handle, signal, jdbiActivityInstance, jdbiActivityInstanceStatus, parentInstanceId);
+
+                if (isNotBlank(targetQuestionStableId)) {
+                    copyAnswerToNewActivityInstance(handle, signal, newActivityInstanceId, sourceAnswers.get(i));
+                }
+            }
+        }
+    }
+
+    /**
+     * Detect an array with child answers inside a composite answer.
+     * Currently supported only composite answers (i.e. `sourceQuestionStableId` - should be
+     * stable_ID of a Composite Question.
+     */
+    private List<Answer> detectSelectedAnswers(Handle handle, long parentInstanceId) {
+        Answer answer = getAnswer(handle, parentInstanceId, sourceQuestionStableId);
+        if (answer instanceof CompositeAnswer) {
+            return getAnswersFromComposite((CompositeAnswer)answer);
+        } else {
+            throw new DDPException("Activity instances creation from answers is supported only for composite questions. StableID="
+                    + sourceQuestionStableId);
+        }
+    }
+
+    /**
+     * Copy answer from source instance to target instance (in target instance should exist a picklist with same
+     * options as in source question, and it should be possible to create an answer with same option stable_id).
+     * We expect than answer - PickListAnswer (i.e. currently we deal only with picklist-inside-composite).
+     *
+     * @param targetInstanceId ID of target activity instance
+     * @param sourceAnswer     source answer (in question `sourceQuestionStableId`) to be copied to question
+     *                         `targetQuestionStableId`
+     */
+    private void copyAnswerToNewActivityInstance(Handle handle, EventSignal signal, long targetInstanceId, Answer sourceAnswer) {
+        var targetAnswer = cloneAnswer(sourceAnswer, sourceQuestionStableId, targetQuestionStableId);
+        var questionDto = getQuestionDto(handle, signal.getStudyId(), targetQuestionStableId);
+        handle.attach(AnswerDao.class).createAnswer(signal.getOperatorId(), targetInstanceId, questionDto.getId(), targetAnswer);
```

Comment:
I think you can call `createAnswer()` without getting `questionDto`, in fact it's probably the way to go here.

Reply:
But how to detect question Id which is required?

Reply:
Each `Answer` object has the `questionStableId`, so `AnswerDao` will figure out the right question id for that activity instance.

Reply:
You mean that it's ok to use this `AnswerDao` method?
` default Answer createAnswer(long operatorId, long instanceId, Answer answer);`

Reply:
Yes.